api/resource.rb

```diff
@@ -250,7 +250,7 @@ def validate
     # rubocop:enable Metrics/MethodLength
 
     def all_user_properties
-      (properties || []) + (parameters || [])
+      ((properties || []) + (parameters || [])).reject(&:exclude)
```

Comment:
Why is this necessary? It looks like we've already taken out rejected properties during the validation phase.

Reply:
I removed them in validate only for Api::Type::NestedObject. Good point though, let me removed them in the validate method of the resource for top level properties too.

Reply:
We can't take out the properties in validate.

`validate` is called prior to applying the overrides. This means that adding an `exclude: true` in `your-provider.yaml` wouldn't have any effect.

The solution would be to perform a second validation after merging the overrides. We have TODOs in the code to do this. However, this is non-trivial in the case of exclude because when we exclude a resource like we do in `terraform.yaml`, the second validation fails because the resource ref points to unexisting resources. In terraform, does resource exist, but they are not generated by Magic Module. The second validation is possible, it would just require some refactoring to disable some checks in specific cases.

For now, it is easier to perform the exclusion of these properties when this method is called.